File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPPotionEffects.java
package org.noxet.noxetserver.playerstate.properties;

import org.bukkit.configuration.ConfigurationSection;
import org.bukkit.entity.Player;
import org.bukkit.potion.PotionEffect;
import org.noxet.noxetserver.playerstate.PlayerStateProperty;

import java.util.Objects;

public class PSPPotionEffects implements PlayerStateProperty<PotionEffect[]> {
    @Override
    public String getConfigName() {
        return "potion_effects";
    }

    @Override
    public PotionEffect[] getDefaultSerializedProperty() {
        return new PotionEffect[0];
    }

    @Override
    public PotionEffect[] getSerializedPropertyFromPlayer(Player player) {
        return player.getActivePotionEffects().toArray(new PotionEffect[0]);
    }

    @Override
    public void restoreProperty(Player player, PotionEffect[] potionEffectList) {
        for(PotionEffect potionEffect : potionEffectList)
            player.addPotionEffect(potionEffect);
    }

    @Override
    @SuppressWarnings("SuspiciousToArrayCall")
    public PotionEffect[] getValueFromConfig(ConfigurationSection config) {
        return Objects.requireNonNull(config.getList(getConfigName())).toArray(new PotionEffect[0]);
    }
}

File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPPotionEffects.scala
package org.noxet.noxetserver.playerstate.properties

import org.bukkit.configuration.ConfigurationSection
import org.bukkit.entity.Player
import org.bukkit.potion.PotionEffect
import org.noxet.noxetserver.playerstate.PlayerStateProperty

import scala.jdk.CollectionConverters.*

object PSPPotionEffects extends PlayerStateProperty[Array[PotionEffect]]:
  override def getConfigName: String = "potion_effects"

  override def getDefaultSerializedProperty: Array[PotionEffect] = Array.empty

  override def getSerializedPropertyFromPlayer(player: Player): Array[PotionEffect] =
    player.getActivePotionEffects.asScala.toArray

  override def restoreProperty(player: Player, potionEffectList: Array[PotionEffect]): Unit =
    potionEffectList.foreach(player.addPotionEffect)

  override def getValueFromConfig(config: ConfigurationSection): Array[PotionEffect] =
    config.getList(getConfigName).asScala.toArray